.github/workflows/goreleaser.yml

```diff
@@ -50,33 +50,42 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - name: Fetch most recent tag, latest (highest version) tag, and second latest tag
+      - name: Set current tag, latest (highest version) tag, and second latest tag
         # For latest and second latest tags, we can't use git tag --sort=version:refname
         # because git doesn't have a concept of pre-release versions and thus mis-sorts
         # versions like 4.0.0-rc.0 *after* 4.0.0.
         run: |
-          echo "most_recent_tag=$(git tag --sort=taggerdate | tail -1)" >> $GITHUB_ENV
+          echo "current_tag=${GITHUB_REF_NAME}" >> $GITHUB_ENV
           echo "latest_tag=$(git tag | tr - \~ | sort --version-sort | tr \~ - | tail -1)" >> $GITHUB_ENV
           echo "second_latest_tag=$(git tag | tr - \~ | sort --version-sort | tr \~ - | tail -2 | sed -n 1p)" >> $GITHUB_ENV
```

Comment:
I'm not sure storing these variables in `$GITHUB_ENV` will work because of how `GITHUB_ENV` works. 
![CleanShot 2022-11-04 at 12 40 25](https://user-images.githubusercontent.com/25608335/199964218-d66db9e2-88e4-472b-92c6-9d56c0b49bf4.png)

It is unique for each step, so it's possible the tags set here won't be accessible in the `Compare tags` step.

Then again the best way to know is by testing it, but I know I've run into this problem in the past that's why I thought to highlight it.

Reply:
Thanks for flagging! Thankfully, I've been able to test this part cross-step successfully -- the items from `Compare tags` and `Log variables` successfully get the values for items set in `GITHUB_ENV` so long as they're part of the same job. Though the wording of this blurb you shared certainly makes me doubt that I had those observations, haha. 😅